pkg/providers/gcp/gcp.go

```diff
@@ -246,19 +172,37 @@ func (c *Client) Refresh(ctx context.Context, project string) {
 				if err != nil {
 					logger.Error("failed to get instance type from url")
 				}
-				c.cache.Set(util.CacheKey(zone, service, name), v1.Instance{
-					Name:    name,
-					Zone:    zone,
-					Service: service,
-					Kind:    kind,
+
+				// Add running instances to the cache
+				cacheKey := util.CacheKey(zone, service, name)
+				instance := &v1.Instance{
+					Provider: provider,
+					Name:     name,
+					Region:   region,
+					Zone:     zone,
+					Service:  service,
```

Comment:
I think shadowing the variable `instance` here might be confusing in the long run and might have unintended consequences (as they are different types)

Also there currently is not really a need to have a new variable here, maybe just pass the `v1.Instance` in its entirety to the `c.cache.Set` function (slightly more memory effcient), or I would say create a variable with a different name